### PR TITLE
plan, executor: implement ExplainInfo() interface for data source operators

### PR DESCRIPTION
**Pull request by zz-jason**
to https://github.com/pingcap/tidb/issues/3805

### CODE REVIEW

File: plan/explain.go [hunk 1]
```diff
@@ -57,6 +57,134 @@ func setParents4FinalPlan(plan PhysicalPlan) {
...
+		}
+	}
+	buffer.WriteString(fmt.Sprintf(", OutOfOrder:%v", p.OutOfOrder))
+	return buffer.String()
+}
+
+// ExplainInfo implements PhysicalPlan interface.
+func (p *physicalTableSource) ExplainInfo() string {
+	buffer := bytes.NewBufferString("")
+	buffer.WriteString(fmt.Sprintf("read only:%v", p.readOnly))
```
Comment (hanfei1991) [inline plan/explain.go]: No need to print readOnly, It will be false forever

File: plan/explain.go [hunk 2]
```diff
@@ -57,6 +57,134 @@ func setParents4FinalPlan(plan PhysicalPlan) {
...
+
+	if p.Aggregated && len(p.aggFuncs) > 0 {
```
Comment (hanfei1991) [inline plan/explain.go]: The following will no longer be used. Needn't print.

Comment (zz-jason) [inline plan/explain.go]: removed

File: plan/explain.go (hunk 1 above)
Comment (zz-jason) [inline plan/explain.go]: removed

File: plan/explain.go [hunk 3]
```diff
@@ -166,7 +245,7 @@ func (p *PhysicalHashJoin) ExplainInfo() string {
 
 // ExplainInfo implements PhysicalPlan interface.
 func (p *PhysicalHashSemiJoin) ExplainInfo() string {
-	buffer := bytes.NewBufferString(fmt.Sprintf("right:%s", p.Children()[1].ID()))
+	buffer := bytes.NewBufferString(fmt.Sprintf("right:%s", p.Children()[p.rightChOffset].ID()))
```
Comment (hanfei1991) [inline plan/explain.go]: Change it and add test....

File: plan/explain.go [hunk 4]
```diff
@@ -207,6 +286,19 @@ func (p *PhysicalMergeJoin) ExplainInfo() string {
 		buffer.WriteString(fmt.Sprintf(", other cond:%s",
 			expression.ExplainExpressionList(p.OtherConditions)))
 	}
+	if len(p.DefaultValues) > 0 {
```
Comment (hanfei1991) [inline plan/explain.go]: remove this....

**Pull request merged**